backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/work/impl/BulkWork.java

```diff
@@ -0,0 +1,195 @@
+	 * <p>
+	 * If {@code true}, no additional refresh of the concerned indexes
+	 * is needed after executing the bulk.
+	 */
+	private final boolean refreshInAPICall;
+
+	protected BulkWork(Builder builder) {
+		super();
+		this.request = builder.buildRequest();
+		this.works = new ArrayList<>( builder.bulkableWorks );
+		this.refreshInAPICall = builder.refreshInBulkAPICall;
+	}
+
+	@Override
+	public String toString() {
+		return new StringBuilder()
+				.append( getClass().getSimpleName() )
+				.append( "[" )
+				.append( "works = " ).append( works )
+				.append( ", refreshInAPICall = " ).append( refreshInAPICall )
+				.append( "]" )
+				.toString();
+	}
+
+	@Override
+	public CompletableFuture<BulkResult> execute(ElasticsearchWorkExecutionContext context) {
+		return Futures.create( () -> context.getClient().submit( request ) )
+				.thenApply( this::generateResult )
+				.exceptionally( Futures.handler(
+						throwable -> {
+							throw log.elasticsearchRequestFailed( request, null, Throwables.expectException( throwable ) ); }
+				) );
+	}
+
+	@Override
+	public void aggregate(ElasticsearchWorkAggregator aggregator) {
+		aggregator.addNonBulkable( this );
+	}
+
+	@Override
+	public Object getInfo() {
+		return null;
+	}
+
+	private BulkResult generateResult(ElasticsearchResponse response) {
+		JsonObject parsedResponseBody = response.getBody();
+		JsonArray resultItems = BULK_ITEMS.get( parsedResponseBody ).orElseGet( JsonArray::new );
+		return new BulkResultDeafualt( resultItems, refreshInAPICall );
+	}
+
+	private static class NoIndexDirtyBulkExecutionContext extends ElasticsearchForwardingWorkExecutionContext {
+
+		public NoIndexDirtyBulkExecutionContext(ElasticsearchWorkExecutionContext delegate) {
+			super( delegate );
+		}
+
+		@Override
+		public void setIndexDirty(URLEncodedString indexName) {
+			// Don't delegate
+		}
+	}
+
+	public static class Builder implements BulkWorkBuilder {
+		private final List<BulkableElasticsearchWork<?>> bulkableWorks;
+		private boolean refreshInBulkAPICall;
+
+		public Builder(List<BulkableElasticsearchWork<?>> bulkableWorks) {
+			this.bulkableWorks = bulkableWorks;
+		}
+
+		@Override
+		public Builder refresh(boolean refresh) {
+			this.refreshInBulkAPICall = refresh;
+			return this;
+		}
+
+		protected ElasticsearchRequest buildRequest() {
+			ElasticsearchRequest.Builder builder =
+					ElasticsearchRequest.post()
+					.pathComponent( Paths._BULK )
+					.param( "refresh", refreshInBulkAPICall );
+
+			for ( BulkableElasticsearchWork<?> work : bulkableWorks ) {
+				builder.body( work.getBulkableActionMetadata() );
+				JsonObject actionBody = work.getBulkableActionBody();
+				if ( actionBody != null ) {
+					builder.body( actionBody );
+				}
+			}
+
+			return builder.build();
+		}
+
+		@Override
+		public BulkWork build() {
+			return new BulkWork( this );
+		}
+	}
+
+	private static class BulkResultDeafualt implements BulkResult {
```

Comment:
There are typos: `BulkResultDeafualt` => `BulkResultDefault`

Also, I'm not sure why it's not simply named `BulkResultImpl`, as in Search 5? Some JQAssistant check were failing maybe?